x/xio/types.go

```diff
@@ -63,11 +82,15 @@ type ReaderSliceOfSlicesIterator interface {
 	// Next moves to the next item
 	Next() bool
 
+	CurrentStart() time.Time
```

Comment:
would it be better to consolidate into a single function?

```
  Next() bool
  
  // Current returns ... 
  Current() (start time.Time, end Time.Time, len int)
```

also, please add comments to the methods